client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -155,6 +156,15 @@
     */
    NotifyingFuture<Boolean> replaceWithVersionAsync(K key, V newValue, long version, int lifespanSeconds, int maxIdleSeconds);
 
+   /**
+    * Retrieve entries from the server
+    *
+    * @param filterConverterFactory Factory name for the KeyValueFilterConverter or null
+    * @param segments               The segments to iterate on or null if all segments should be iterated
+    * @param batchSize              The number of entries transferred from the server at a time
+    * @return Iterator for the entries
+    */
+   CloseableIterator<Entry<Object, Object>> retrieveEntries(String filterConverterFactory, Set<Integer> segments, int batchSize);
```

Comment:
I think we need to think how to expose the segment info in org.infinispan.client.hotrod.impl.consistenthash.ConsistentHash / SegmentConsistentHash to the client.